components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -235,4 +235,59 @@ public String getName() {
 
         return APIConstants.GatewayArtifactSynchronizer.DB_RETRIEVER_NAME;
     }
+
+    @Override
+    public String retrieveGatewayPolicyArtifacts(String mappingUUID) throws ArtifactSynchronizerException {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        if (gatewayArtifactSynchronizerProperties.hasEventWaitingTime()) {
+            try {
+                Thread.sleep(gatewayArtifactSynchronizerProperties.getEventWaitingTime());
+            } catch (InterruptedException e) {
+                log.error("Error occurred while waiting to retrieve artifacts from event hub");
+            }
+        }
+        try {
+            String path = APIConstants.GatewayArtifactSynchronizer.GATEWAY_POLICY_SYNAPSE_ARTIFACTS +
+                    "?policyMappingUuid=" + mappingUUID + "&type=Synapse";
+            String endpoint = baseURL + path;
+            try (CloseableHttpResponse httpResponse = invokeService(endpoint, tenantDomain)) {
+                JSONArray jsonArray = retrieveArtifact(httpResponse);
+                if (jsonArray != null && jsonArray.length() > 0) {
+                    return jsonArray.getString(0);
+                }
+            }
+        } catch (IOException e) {
+            String msg = "Error while executing the http client";
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
+        }
+        return null;
+    }
+
+    @Override
+    public List<String> retrieveGatewayPolicyArtifacts(String gatewayLabel, String tenantDomain)
+            throws ArtifactSynchronizerException {
+        List<String> gatewayPolicyArtifactsArray = new ArrayList<>();
+        try {
+            String encodedGatewayLabel = URLEncoder.encode(gatewayLabel, APIConstants.DigestAuthConstants.CHARSET);
+            String path = APIConstants.GatewayArtifactSynchronizer.GATEWAY_POLICY_SYNAPSE_ARTIFACTS
+                    + "?gatewayLabel=" + encodedGatewayLabel + "&type=Synapse";
+            String endpoint = baseURL + path;
+            try (CloseableHttpResponse httpResponse = invokeService(endpoint,tenantDomain)) {
+                JSONArray jsonArray = retrieveArtifact(httpResponse);
+                if (jsonArray != null) {
+                    for (int i = 0; i < jsonArray.length(); i++) {
+                        gatewayPolicyArtifactsArray.add(jsonArray.getString(i));
+                    }
+                }
+
```

Comment:
remove additional line

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/1136a795360198388c6b4bd97029c20b0f898832

Thanks and Regards